From e8ff27d6a29550d997372f41249dac3f83668b5b Mon Sep 17 00:00:00 2001
From: Jianhong Yin <yin-jianhong@163.com>
Date: Sat, 21 Aug 2021 13:52:57 -0400
Subject: [PATCH 11/11] nfs-utils: add install-dep for installing all
 dependencies

Whenever user want to compile and install from source code, they
have to constantly install dependencies based on error message.

Verified on RHEL-8/Fedora-34/debian-10/openSUSE-15.3

Signed-off-by: Jianhong Yin <yin-jianhong@163.com>
Signed-off-by: Steve Dickson <steved@redhat.com>
---
 install-dep | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100755 install-dep

diff --git a/install-dep b/install-dep
new file mode 100755
index 00000000..621618fe
--- /dev/null
+++ b/install-dep
@@ -0,0 +1,21 @@
+#!/bin/bash
+#install dependencies for compiling from source code
+
+#RHEL/Fedora/CentOS-Stream/Rocky
+which dnf &>/dev/null || which yum &>/dev/null && {
+	yum install -y automake libtool make gcc rpcgen libtirpc-devel libevent-devel sqlite-devel device-mapper-devel \
+		libblkid-devel krb5-devel libuuid-devel
+}
+
+#Debian/ubuntu
+which apt &>/dev/null && {
+	apt install -o APT::Install-Suggests=0 -o APT::Install-Recommends=0 --ignore-missing -y \
+		autotools-dev automake make libtool pkg-config libtirpc-dev libevent-dev libsqlite3-dev \
+		libdevmapper-dev libblkid-dev libkrb5-dev libkeyutils-dev uuid-dev
+}
+
+#openSUSE Leap
+which zypper &>/dev/null && {
+	zypper in --no-recommends -y automake libtool make gcc rpcgen libtirpc-devel libevent-devel sqlite-devel \
+		device-mapper-devel libblkid-devel krb5-devel libuuid-devel
+}
-- 
2.33.0

